components/select/select-label-factory.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { ReactElement } from 'react';
+import {
+	Container,
+	Padding,
+	Text,
+	Row,
+	Icon,
+	TextProps,
+	IconProps,
+	SelectItem,
+	LabelFactoryProps,
+	getColor
+} from '@zextras/carbonio-design-system';
+import styled from 'styled-components';
+
+export const Square = styled.div`
+	width: 1rem;
+	height: 1rem;
+	background: ${({ color }): string | undefined => color};
+	border-radius: 0.25rem;
+	opacity: ${({ disabled }: { disabled?: boolean }): number => (disabled ? 0.5 : 1)};
+`;
+
+export const ColorContainer = styled(Container)`
+	border-bottom: 0.0625rem solid ${({ theme }): string => theme.palette.gray2.regular};
+	transition: background 0.2s ease-out;
+	&:hover {
+		background: ${({ theme, background }): string => getColor(`${background}.hover`, theme)};
+	}
+`;
+
+export const TextUpperCase = styled(Text)`
+	text-transform: capitalize;
+	color: ${({ theme, disabled }): string =>
+		disabled ? theme.palette.text.disabled : theme.palette.text.regular};
+`;
+export const LabelText = styled(Text)<TextProps & { $showPrimary?: boolean }>`
+color: ${({ theme, disabled, $showPrimary }): string =>
+	// eslint-disable-next-line no-nested-ternary
+	disabled
+		? theme.palette.text.disabled
+		: $showPrimary
+		? theme.palette.primary.regular
+		: theme.palette.secondary.regular}};`;
+
+export const StyledIcon = styled(Icon)<IconProps & { $showPrimary?: boolean }>`
+color: ${({ theme, disabled, $showPrimary }): string =>
+	// eslint-disable-next-line no-nested-ternary
+	disabled
+		? theme.palette.text.disabled
+		: $showPrimary
+		? theme.palette.primary.regular
+		: theme.palette.secondary.regular}};`;
+
+interface CustomSelectItem extends SelectItem {
```

Comment:
I think there is redundancy with the types. Actually only this interface can be exported as type (in the d.ts) instead of the _SelectItems_ type and used both in the label factory and in the selector component.
Also the names should be changed to be more specific like _FolderSelectorItem_, _FolderSelectorLabelFactory_, etc...